### PR TITLE
Module 1: Core concepts

### PR DESCRIPTION
**Pull request by bioN1ck**

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,26 +1,33 @@
...
+import './App.scss';
+import Counter from './components/Counter/Counter';
+import Search from './components/Search/Search';
+import ResultsFilter from './components/ResultsFilter/ResultsFilter';
+import { Genres } from './models/genres.model';
 
-function App() {
-  return (
-    <div className="App">
+class App extends React.Component {
```
Comment (melk0sha) [inline src/App.tsx]: It's better to use functional components in React:
https://react.dev/reference/react/Component#migrating-a-simple-component-from-a-class-to-a-function
https://react.dev/reference/react/Component#defining-a-class-component (see first "Pitfall")

Except of `Counter` component as it is a requirement to use class syntax

File: src/App.tsx [hunk 2]
```diff
@@ -1,26 +1,33 @@
...
+
+  private genres = Object.values(Genres).map((genre, id) => ({ id, genre }));
+
+  render() {
+    return <div className="App">
```
Comment (melk0sha) [inline src/App.tsx]: For CSS, there are naming conventions about classNames:
https://www.freecodecamp.org/news/css-naming-conventions-that-will-save-you-hours-of-debugging-35cea737d849/

File: src/components/ResultsFilter/ResultsFilter.tsx [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+}
+
+function ResultsFilter(props: ResultsFilterProps) {
+  const [state, setState] = useState(props);
+
+  const handleGenreSelect = (genre: Genres): void => {
+    state.onSelect(genre);
+    setState({ ...state, activeGenre: genre });
+  };
+  const checkActiveness = (genre: Genres): string => {
```
Comment (melk0sha) [inline src/components/ResultsFilter/ResultsFilter.tsx]: For such cases in future you can use one of the following packages that will help you to easy define customizible classNames:
https://www.npmjs.com/package/classnames
https://www.npmjs.com/package/clsx

File: src/components/ResultsFilter/ResultsFilter.tsx [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+import { Genre, Genres } from '../../models/genres.model';
+
+
+type ResultsFilterProps = {
+  genres: Genre[];
+  activeGenre: Genres;
+  onSelect: (genre: Genres) => void;
+}
+
+function ResultsFilter(props: ResultsFilterProps) {
```
Comment (melk0sha) [inline src/components/ResultsFilter/ResultsFilter.tsx]: I believe this component and the whole file should be named `GenreSelect`

File: src/components/Search/Search.tsx [hunk 5]
```diff
@@ -0,0 +1,34 @@
+import { KeyboardEvent, MutableRefObject, useRef } from 'react';
+import './Search.scss';
+
+type SearchProps = {
+  initialValue: string;
+  onSearch: (val: string) => void;
+}
+
+function Search({ initialValue, onSearch }: SearchProps) {
+  const inputRef: MutableRefObject<HTMLInputElement | null> = useRef(null);
```
Comment (melk0sha) [inline src/components/Search/Search.tsx]: You can simply use here such typing:
` const inputRef = useRef<HTMLInputElement>(null);`

**Review by melk0sha — CHANGES_REQUESTED** [review]

**Review by melk0sha — APPROVED** [review]

**Comment by melk0sha** [comment]
Also, I don't know if you are using `Prettier` but I suggest you to install and use it because it saves time by code formatting. It can be managed to format the code by file saving.
You can add it as an extension for your IDE 🙂
https://prettier.io/

**Review by melk0sha — APPROVED** [review]

File: src/App.tsx (hunk 1 above)
Comment (bioN1ck) [reply src/App.tsx]: Fixed

File: src/App.tsx (hunk 2 above)
Comment (bioN1ck) [reply src/App.tsx]: Switched to BEM

File: src/components/ResultsFilter/ResultsFilter.tsx (hunk 3 above)
Comment (bioN1ck) [reply src/components/ResultsFilter/ResultsFilter.tsx]: Used the classnames pkg

File: src/components/ResultsFilter/ResultsFilter.tsx (hunk 4 above)
Comment (bioN1ck) [reply src/components/ResultsFilter/ResultsFilter.tsx]: Fixed

File: src/components/Search/Search.tsx (hunk 5 above)
Comment (bioN1ck) [reply src/components/Search/Search.tsx]: Used proposed

**Pull request merged**